src/rebar_dir.erl

```diff
@@ -144,7 +144,7 @@ extra_src_dirs(State) -> extra_src_dirs(State, []).
 extra_src_dirs(State, Default) ->
     ErlOpts = rebar_utils:erl_opts(State),
     Vs = proplists:get_all_values(extra_src_dirs, ErlOpts),
-    case lists:append(Vs) of
+    case lists:append(Vs) ++ rebar_state:get(State, extra_src_dirs, []) of
```

Comment:
`lists:append([rebar_state:get(State, extra_src_dirs, []) | Vs])` will be faster.